File: 27/polyakov/r00/solve27.py
Fin = open("27.txt")

N = int( Fin.readline() )

s, dMin  = 0, 10001
for i in range(N):
  a, b = map( int, Fin.readline().split() )
  s += max( a, b )
  d = abs( a-b )
  if d % 3 > 0:
    dMin = min( d, dMin )

if s % 3 != 0:
  print( s )
else:
  print( s-dMin )

Fin.close()

File: 27/polyakov/r00/solve27.pas
var N, i, a, b : integer;
begin
  Assign(input, '27-b.txt');
  Readln( N );
  var s := 0;
  var dMin := MaxInt;
  for i:=1 to N do begin
    Readln( a, b );
    s += Max(a,b);
    var d := Abs(a-b);
    if d mod 3 <> 0 then
      dMin := Min(d, dMin)
  end;
  if s mod 3 <> 0 then
       Writeln( s )
  else Writeln( s-dMin );
end.
